### PR TITLE
feat(landing): Add new features pop up to alert users for recent releases. BM-1064

### PR DESCRIPTION
**Pull request by Wentao-Kuang**
### Motivation
We want to alert Basemaps users about new features for recent release. To use the @linzjs/lui -> [LuiUpdatesSplashModal](https://github.com/linz/Lui/blob/master/src/components/LuiUpdateSplashModal/LuiUpdatesSplashModal.tsx) which is same with land online search.

### Modifications
Initial investigations with @linzjs/lui that it's very hard to import and use react components from it because different tech stacks.
- lui build and bundled with nodejs which contains dependencies like `crypto` or `stream` that is not support by browser. We might need to do polyfills to map them into browser friendly dependencies.
- lui only bundle everything into one index.js, we don't have individual module file to do direct import, so we have to deal with dependency problem above.

As a result, we decide to re-implement the `LuiUpdatesSplashModal` in basemaps just using the assets and css from @linzjs/lui
- `FeatureUpdates` is a completely re-implementation of `LuiUpdatesSplashModal`
- `NewFeature` is the component that we need to update often when describe new features.
- Updated @linzjs/lui to latest

### Verification
- Tested new pop up only show once after dismissed.
- Tested new pop up stop showing after closing date.
- Tested new pop up shows in incognito mode/clearing cache after dismissed.
image
image

### CODE REVIEW

File: packages/landing/src/components/feature.updates.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  handleClose = (): void => {
+    this.setState({ showModal: false });
+    window.localStorage.setItem(this.props.id, this.props.releaseVersion);
+  };
+
+  renderFeatureMedia(): ReactNode {
+    const { bigImage, smallImage, iframe } = this.props;
+    if (iframe) {
```
Comment (blacha) [inline packages/landing/src/components/feature.updates.tsx:52]: we have no plans for the iframe variant, could we rip all this out?

File: packages/landing/src/components/new.feature.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Config } from '../config.js';
+import { WindowUrl } from '../url.js';
+import { FeatureUpdates } from './feature.updates.js';
+
+const baseUrl = WindowUrl.baseUrl();
+
+/**
+ * Please updated the following settings and descriptions for new features pop up
+ */
+const bigImage = new URL('assets/Lg+3D+Maps+splash.gif', baseUrl).href; // Large gif file location
```
Comment (blacha) [inline packages/landing/src/components/new.feature.tsx:12]: could this be turned into a new.feature.3dmap.tsx or similar ?

so we have a component per new feature

File: packages/landing/src/components/feature.updates.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+ * This use to enable a one off pop up screen for introduce new feature of the recent release.
+ *
+ */
+export class FeatureUpdates extends Component<FeatureUpdatesProps, FeatureUpdatesState> {
+  constructor(props: FeatureUpdatesProps) {
+    super(props);
+    const currentVersion = window.localStorage.getItem(this.props.id);
+
+    this.state = {
+      showModal: this.props.enabled && this.props.releaseVersion.trim() !== currentVersion,
```
Comment (blacha) [inline packages/landing/src/components/feature.updates.tsx:41]: do we have a requirement that releaseversion limits when this is shown?

Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx:41]: Not in the Acceptance Criteria. But I think [STEP is using split io](https://github.com/linz/landonline-search/blob/c1f02a4e2ec427c211896495bea4a886bc92f7ea/web/src/components/RecentUpdates.tsx#L15) for the control. We only limited by an end date.

Comment (blacha) [reply packages/landing/src/components/feature.updates.tsx:41]: version is not a date?

Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx:41]: The `releaseVersion` basically just a cache of `dissmissKey` in lui.
I have put the date check in the `[new.feature.tsx](https://github.com/linz/basemaps/pull/3356/files#diff-ed4cda0d05e45a43654131c1ff2f93f88f4c0a7e7f7e84471b1b52d9c8b94077)` line 32.

File: packages/landing/src/components/new-features/3d.map.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+const recentUpdates = {
+  children: (
+    <>
+      <h5 className="RecentUpdatesHeading">Basemaps are now viewable in 3D!</h5>
+      <p>
+        To activate this function, click the mountains icon on the left-hand side then hold right-click to change your
+        viewpoint.
+      </p>
+      <p>The new Labels button can also be toggled to show places names.</p>
+    </>
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply packages/landing/src/components/new-features/3d.map.tsx:25]: I am getting this h5 from [landonline-title](https://github.com/linz/landonline-titles/blob/6cf267b39a4232d13d6e1e32f9d2dd7630795f73/web/src/modules/landing/components/RecentSplashScreenUpdates/RecentSplashScreenUpdates.tsx#L4).

But seems h2 looks better for me.

File: packages/landing/src/components/feature.updates.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  // @linzjs/lui cross_icon re-implementation
+  ClearIcon(): ReactNode {
+    return (
+      <span className="LuiIcon LuiIcon--md LuiIcon--interactive" data-icon="ic_clear" aria-label="cross_icon">
+        <img src="/assets/clear.svg" alt="cross_icon" />
+      </span>
+    );
+  }
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx:105]: Agree, I think use `img` might be not best solution, but I don't like that we just hard code `svg` in the typescript neither. As we only bundled svg files from lui into landing/assets, we might need a better to import svg with a path.

File: packages/landing/src/components/feature.updates.tsx (hunk 1 above)
Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx]: All removed

File: packages/landing/src/components/new.feature.tsx (hunk 2 above)
Comment (Wentao-Kuang) [reply packages/landing/src/components/new.feature.tsx]: Updated.

File: packages/landing/src/components/feature.updates.tsx (hunk 3 above)
Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx]: I have removed the releaseVersion and only add `dissmissKey` and `closeDate` for now.

File: packages/landing/src/components/feature.updates.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    return (
+      <ReactModal
+        isOpen={showModal}
+        shouldCloseOnEsc={true}
+        onRequestClose={this.handleClose}
+        shouldCloseOnOverlayClick={true}
+        contentLabel="Recent updates"
+        className="lui-splash-content lui-box-shadow"
+        overlayClassName="splash_overlay"
+      >
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx:59]: Fixed, thank you.

File: packages/landing/src/components/feature.updates.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply packages/landing/src/components/feature.updates.tsx:106]: We might fix this later.

File: packages/landing/src/components/new-features/3d.map.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply packages/landing/src/components/new-features/3d.map.tsx]: Updated to h2.

**Pull request merged**